github/checks_test.go

```diff
@@ -104,25 +98,19 @@ func TestChecksService_GetCheckSuite(t *testing.T) {
 		t.Errorf("Checks.GetCheckSuite return %+v, want %+v", checkSuite, want)
 	}
 
-	// Test addOptions failure
-	_, _, err = client.Checks.GetCheckSuite(ctx, "\n", "\n", -1)
-	if err == nil {
-		t.Error("bad options GetCheckSuite err = nil, want error")
-	}
+	const methodName = "GetCheckSuite"
+	testBadOptions(t, methodName, func() (err error) {
+		_, _, err = client.Checks.GetCheckSuite(ctx, "\n", "\n", -1)
+		return err
+	})
 
-	// Test s.client.Do failure
-	client.BaseURL.Path = "/api-v3/"
-	client.rateLimits[0].Reset.Time = time.Now().Add(10 * time.Minute)
-	got, resp, err := client.Checks.GetCheckSuite(ctx, "o", "r", 1)
-	if got != nil {
-		t.Errorf("rate.Reset.Time > now Getchecksuite = %#v, want nil", got)
-	}
-	if want := http.StatusForbidden; resp == nil || resp.Response.StatusCode != want {
-		t.Errorf("rate.Reset.Time > now Getchecksuite resp = %#v, want StatusCode=%v", resp.Response, want)
-	}
-	if err == nil {
-		t.Error("rate.Reset.Time > now Getchecksuite err = nil, want error")
-	}
+	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
+		got, resp, err := client.Checks.GetCheckSuite(ctx, "o", "r", 1)
+		if got != nil {
+			t.Errorf("rate.Reset.Time > now %v = %#v, want nil", methodName, got)
```

Comment:
Just skimmed the changes so far, and noticed that the error messages for the function passed to `testNewRequestAndDoFailure` seem to be inconsistent in how they are worded.

Reply:
Ah, thank you!  Fixed.